cmd/storj/cmd/ls.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package cmd
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	"github.com/minio/minio/pkg/auth"
+	"github.com/minio/minio/pkg/hash"
+	"github.com/spf13/cobra"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/cfgstruct"
+	"storj.io/storj/pkg/process"
+)
+
+var (
+	lsCfg Config
+	lsCmd = &cobra.Command{
+		Use:   "ls",
+		Short: "A brief description of your command",
+		RunE:  list,
+	}
+)
+
+func init() {
+	RootCmd.AddCommand(lsCmd)
+	cfgstruct.Bind(lsCmd.Flags(), &lsCfg, cfgstruct.ConfDir(defaultConfDir))
+	lsCmd.Flags().String("config", filepath.Join(defaultConfDir, "config.yaml"), "path to configuration")
+}
+
+func list(cmd *cobra.Command, args []string) error {
+	ctx := process.Ctx(cmd)
+
+	if len(args) == 0 {
+		return errs.New("No file specified for copy")
+	}
+
+	if len(args) == 1 {
+		return errs.New("No destination specified")
+	}
+
+	//TODO: actually get the proper config
+	identity, err := lsCfg.LoadIdentity()
+	if err != nil {
+		return err
+	}
+
+	gateway, err := lsCfg.NewGateway(ctx, identity)
+	if err != nil {
+		return err
+	}
+
+	credentials, err := auth.CreateCredentials(lsCfg.AccessKey, lsCfg.SecretKey)
+	if err != nil {
+		return err
+	}
+
+	storjObjects, err := gateway.NewGatewayLayer(credentials)
```

Comment:
since you're creating this in two commands, it might be nice to make a helper function?